File: 23-1/programmers/실패율.py
def solution(N, stages):

    peoples = len(stages)
    stages.sort()
    fail = [ 0 for _ in range(N+1)]
    for idx in stages:
        if idx==N+1:
            continue
        fail[idx]+=1
    stage_rate = []
    fail = fail[1:]
    for idx, count in enumerate(fail):
        fail_players = fail[idx]
        stage_players = peoples-sum(fail[:idx])
        if stage_players == 0:
            rate = 0
        else:
            rate = fail_players/stage_players
        stage_rate.append([rate, idx+1])
    stage_rate.sort(key=lambda x : (-x[0] ))
    answer = []
    for rate, num in stage_rate:
        answer.append(num)
    return answer

File: 프로그래머스/1/42889. 실패율/실패율.js
function solution(N, stages) {
    const count = new Array(N+2).fill(0)
    let total = stages.length
    const failRate= []
    for(let s of stages){
        count[s]+=1
    }
    for (let idx=1;idx<count.length-1;idx++){
        //console.log(count[idx],total)
        failRate.push([count[idx]/total,idx])
        total-=count[idx]
       
    }
 
    failRate.sort((a,b)=>(-a[0]+b[0]))
    //console.log(failRate)
    const answer = []
    for(let e of failRate){
        answer.push(e[1])
    }

    return answer;
}